ocaml/typing/typedecl.ml

```diff
@@ -2170,6 +2172,20 @@ let get_native_repr_attribute attrs ~global_repr =
   | _, Some { Location.loc }, _ ->
     raise (Error (loc, Multiple_native_repr_attributes))
 
+let is_upstream_compatible_non_value_unbox env ty =
```

Comment:
When we have a better treatment for unboxed type names (which should store the base name and the `#` separately), this should change. Please leave a CR (v2.5).

Reply:
added CR